src/bytes.rs

```diff
@@ -265,19 +275,48 @@ impl<'a, T: Sized + AsRef<[u8]> + Send + Sync + 'a> From<Arc<T>> for Bytes<'a> {
     }
 }
 #[cfg(feature = "alloc")]
+impl From<alloc::boxed::Box<[u8]>> for Bytes<'_> {
+    fn from(value: alloc::boxed::Box<[u8]>) -> Self {
+        unsafe extern "C" fn release(this: *const (), capacity: usize) {
+            core::mem::drop(alloc::boxed::Box::from_raw(
+                core::ptr::slice_from_raw_parts_mut(this.cast::<u8>().cast_mut(), capacity),
+            ))
+        }
+        let data: &[u8] = value.as_ref();
```

Comment:
Let's try to be extra careful with provenance, aliasing, and miri being pedantic. Let's thus "disarm" the (lack of) aliasing _validity_ invariant of `Box` before we start aliasing it:

```suggestion
        let data = alloc::boxed::Box::into_raw(value).cast::<()>();
        let bytes: &[u8] = unsafe { &*data };
```